spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -148,7 +148,7 @@ public boolean next() throws IOException {
       if (currentTask != null && !currentTask.isDataTask()) {
         String filePaths =
             referencedFiles(currentTask)
-                .map(file -> file.path().toString())
+                .map(file -> file.location())
```

Comment:
Spark v3.5 uses method reference here. Feel free to ignore this and other comments about method reference if it's intentional.

Reply:
Let's keep it consistent, I'll fix it!